python/pylibcudf/pylibcudf/io/types.pyx

```diff
@@ -37,13 +42,233 @@ __all__ = [
     "CompressionType",
     "DictionaryPolicy",
     "JSONRecoveryMode",
+    "PartitionInfo",
     "QuoteStyle",
     "SinkInfo",
     "SourceInfo",
     "StatisticsFreq",
+    "TableInputMetadata",
     "TableWithMetadata",
 ]
 
+cdef class PartitionInfo:
+    """
+    Information used while writing partitioned datasets.
+
+    Parameters
+    ----------
+    start_row : int
+        The start row of the partition.
+
+    num_rows : int
+        The number of rows in the partition.
+    """
+    def __init__(self, int start_row, int num_rows):
```

Comment:
Are we limited to files with < 2B rows? Maybe `int` isn't right here either. Check what libcudf does.

Reply:
Yeah appears libcudf uses `size_type` so changed it to that